File: app/src/main/java/com/example/smoiapp001/loaders/LoadPopularCost.java
package com.example.smoiapp001.loaders;

import android.content.Context;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.loader.content.AsyncTaskLoader;

import com.example.smoiapp001.database.AppDatabase;

public class LoadPopularCost extends AsyncTaskLoader<Float> {
    private AppDatabase db;
    private String keyword;

    public LoadPopularCost(@NonNull Context context, AppDatabase db, String keyword ) {
        super(context);
        this.db = db;
        this.keyword = keyword;
    }

    @Override
    protected void onStartLoading() {
        super.onStartLoading();
        forceLoad();
    }

    @Nullable
    @Override
    public Float loadInBackground() {
        return db.transactionDao().loadRecommendedCostByDescription(keyword);
    }

}

File: app/src/main/java/com/example/smoiapp001/loaders/LoadPopularCost.kt
package com.example.smoiapp001.loaders

import android.content.Context
import androidx.loader.content.AsyncTaskLoader

import com.example.smoiapp001.database.AppDatabase

class LoadPopularCost(context: Context, private val db: AppDatabase, private val keyword: String) : AsyncTaskLoader<Float>(context) {

    override fun onStartLoading() {
        super.onStartLoading()
        forceLoad()
    }

    override fun loadInBackground(): Float? {
        return db.transactionDao().loadRecommendedCostByDescription(keyword)
    }

}
